### PR TITLE
Feature/sidechain registration

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
### Description of the Change
Registration based on sidechain event

### CODE REVIEW

File: notary-commons/src/main/kotlin/com/d3/commons/registration/ChainRegistrationWithProvidedAddress.kt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    private val ethereumAccountRegistrator = SideChainRegistrator(
+        irohaConsumer,
+        storageAccountId,
+        chainKeyName
+    )
+
+    /**
+     * @param accountId - account that will be assigned a provided ethereum address
+     * @param ethAddress - ethereum address for iroha client with clientId
+     * @return transaction of
```
Comment (dolgopolovwork) [inline notary-commons/src/main/kotlin/com/d3/commons/registration/ChainRegistrationWithProvidedAddress.kt]: `transaction of` what? And I see no function description. Just `@param`s

File: notary-commons/src/main/kotlin/com/d3/commons/registration/ChainRegistrationWithProvidedAddress.kt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    fun register(
+        accountId: String,
+        chainAddress: String,
+        time: BigInteger
+    ): IrohaTransaction {
+        val assignedAddress = chainAddressProvider.getAddressByAccountId(accountId).get()
+        if (assignedAddress.isPresent)
+            throw D3ErrorException.warning(
+                failedOperation = REGISTRATION_OPERATION,
+                description = "Client $accountId has already been registered with $chainKeyName address: ${assignedAddress.get()}"
```
Comment (Mingela) [inline notary-commons/src/main/kotlin/com/d3/commons/registration/ChainRegistrationWithProvidedAddress.kt]: isn't it a confidential information?

File: notary-commons/src/main/kotlin/com/d3/commons/registration/ChainRegistrationWithProvidedAddress.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                    chainAddress
+                ),
+                IrohaCommand.CommandSetAccountDetail(
+                    storageAccountId,
+                    chainAddress,
+                    accountId
+                )
+            )
+        )
+
```
Comment (Mingela) [inline notary-commons/src/main/kotlin/com/d3/commons/registration/ChainRegistrationWithProvidedAddress.kt]: :suspect:

**Review by dolgopolovwork — APPROVED** [review]

File: notary-commons/src/main/kotlin/com/d3/commons/registration/ChainRegistrationWithProvidedAddress.kt [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            time,
+            irohaConsumer.getConsumerQuorum().get(),
+            arrayListOf(
+                IrohaCommand.CommandSetAccountDetail(
+                    accountId,
+                    chainKeyName,
+                    chainAddress
+                ),
+                IrohaCommand.CommandSetAccountDetail(
+                    storageAccountId,
```
Comment (Mingela) [inline notary-commons/src/main/kotlin/com/d3/commons/registration/ChainRegistrationWithProvidedAddress.kt:61]: `storageAccountId` -> `sideChainAddressesStorage` ?

File: notary-commons/src/test/kotlin/com/d3/commons/registration/ChainRegistrationWithProvidedAddressTest.kt [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    private val unassignedClientid = "unassigned@client"
+    private val chainAddress = "0xchain_address"
+    private val storageAccountId = "storage@notary"
+    private val chainId = "chain_id"
+    private val time = BigInteger.ZERO
+
+    private val ethAddressProvider = mock<ChainAddressProvider> {
+        on {
+            getAddressByAccountId(
+                eq(assignedClientId)
```
Comment (Mingela) [inline notary-commons/src/test/kotlin/com/d3/commons/registration/ChainRegistrationWithProvidedAddressTest.kt]: weird formatting imo

**Review by Mingela — APPROVED** [review]

**Pull request merged**